processing/src/main/java/io/druid/segment/serde/FloatGenericColumnPartSerdeV2.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.serde;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.druid.collections.bitmap.ImmutableBitmap;
+import io.druid.java.util.common.io.smoosh.FileSmoosher;
+import io.druid.segment.column.ValueType;
+import io.druid.segment.data.BitmapSerde;
+import io.druid.segment.data.BitmapSerdeFactory;
+import io.druid.segment.data.CompressedColumnarFloatsSupplier;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.ByteOrder;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ */
+public class FloatGenericColumnPartSerdeV2 implements ColumnPartSerde
```

Comment:
Ensure that Double, Float and Long have the same method order (and code) and differ only by the type

Reply:
changed method order to existing order in FloatGenericColumnPartSerde and 
also compared each other they are same. 
also, if you think same method ordering is really required and we need to enforce that, ould be great if we can add a checkstyle check for this. 

IMO, making the individual method order same in different classes is not something that need not to be enforced.

Reply:
It couldn't be enforced easily, also those specializations need to have some minor differences (apart from the type) sometimes.

Reply:
If you really think having same method ordering across multiple classes is valuable and worth a developer investing his time in rearranging methods and comparing multiple classes, would be great if you can add this to contribution guidelines on druid docs. 

Would really help in decreasing back and forth on future PRs and decrease the amount of iterations we need to go through w.r.t method orderings, which IMO do not add any value to the feature being added.

Reply:
Created - https://github.com/druid-io/druid/issues/5407